client/blocks/subscriptions-module-banner/index.tsx

```diff
@@ -1,72 +1,78 @@
 import { Notice } from '@wordpress/components';
+import { usePrevious } from '@wordpress/compose';
 import { useTranslate } from 'i18n-calypso';
-import { useEffect, useRef, useCallback } from 'react';
+import { useEffect, useCallback } from 'react';
 import QueryJetpackModules from 'calypso/components/data/query-jetpack-modules';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
-import useBannerDismissReducer from 'calypso/my-sites/people/subscribers/hooks/use-banner-dismiss-reducer';
-import useSubscriptionBanner from 'calypso/my-sites/people/subscribers/hooks/use-subscription-banner';
 import { getFormSettings } from 'calypso/my-sites/site-settings/form-discussion';
 import { useDispatch, useSelector } from 'calypso/state';
 import { activateModule } from 'calypso/state/jetpack/modules/actions';
 import { saveJetpackSettings } from 'calypso/state/jetpack/settings/actions';
 import getJetpackSettings from 'calypso/state/selectors/get-jetpack-settings';
 import isJetpackModuleActive from 'calypso/state/selectors/is-jetpack-module-active';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
+import useBannerDismissReducer from './use-banner-dismiss-reducer';
+import useSubscriptionBanner from './use-subscription-banner';
 
-export default function BannerActivation() {
-	const moduleSlug = 'subscriptions';
+import './style.scss';
+
+const moduleSlug = 'subscriptions';
+
+export default function SubscriptionsModuleBanner() {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
 
 	const site = useSelector( ( state ) => getSelectedSite( state ) );
-	const siteId = site?.ID as number;
-	const siteJetpackSettings = useSelector( ( state ) => getJetpackSettings( state, siteId ) );
-	const isModuleActive = useSelector( ( state ) =>
-		isJetpackModuleActive( state, siteId, moduleSlug )
-	);
-	const isModuleActiveRef = useRef( isModuleActive );
+	const siteId = site?.ID;
+	const siteJetpackSettings = useSelector( ( state ) => {
+		if ( ! siteId ) {
+			return null;
+		}
+		return getJetpackSettings( state, siteId );
+	} );
+	const isModuleActive = useSelector( ( state ) => {
+		if ( ! siteId ) {
+			return null;
+		}
+		return isJetpackModuleActive( state, siteId, moduleSlug );
+	} );
+	const prevIsModuleActive = usePrevious( isModuleActive );
 	const [ bannerState, bannerDispatch ] = useBannerDismissReducer();
 	const showSubscriptionBanner = useSubscriptionBanner( siteId, bannerState.dismissed );
 
 	/**
 	 * Callbacks
 	 */
-	const prepareModuleSettings = useCallback( () => {
-		return Object.assign( {}, getFormSettings( siteJetpackSettings ), {
-			stb_enabled: true,
-			stc_enabled: true,
-		} );
-	}, [ getFormSettings, siteJetpackSettings ] );
-
 	const onEnableSubscriptionsModule = useCallback( () => {
-		dispatch( activateModule( siteId, moduleSlug ) );
-		recordTracksEvent( 'calypso_subscriptions_module_enable' );
-	}, [ siteId ] );
+		if ( typeof siteId === 'number' ) {
```

Comment:
Is this too strict or is it guaranteed it's going to a number for sure?

Reply:
Based on the TypeScript types, it's expected to be a number. My local tests of the PR also confirmed this.

Reply:
Okay.